modules/basics/src/main/java/com/opengamma/basics/date/LocalDateUtils.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.basics.date;
+
+import java.time.LocalDate;
+
+/**
+ * Utilities for working with {@code LocalDate}.
+ */
+final class LocalDateUtils {
+
+  // First day-of-month minus one for a standard year
+  private static final int[] STANDARD = {0, 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334};
+  // First day-of-month minus one for a leap year
+  private static final int[] LEAP = {0, 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335};
+
+  /**
+   * Restricted constructor.
+   */
+  private LocalDateUtils() {
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Finds the day-of-year of the date.
+   * <p>
+   * Faster than the JDK method.
+   * 
+   * @param date  the date to query
+   * @return the day-of-year
+   */
+  static int doy(LocalDate date) {
+    int[] lookup = (date.isLeapYear() ? LEAP : STANDARD);
+    return lookup[date.getMonthValue()] + date.getDayOfMonth();
+  }
+
+  /**
+   * Adds a number of days to the date.
+   * <p>
+   * Faster than the JDK method.
+   * 
+   * @param date  the date to add to
+   * @return the new date
+   */
+  static LocalDate plusDays(LocalDate date, int daysToAdd) {
+    if (daysToAdd == 0) {
+      return date;
+    }
+    long dom = date.getDayOfMonth() + daysToAdd;
+    if (dom > 0 && dom <= 59) { // 59th Jan is 28th Feb, 59th Feb is 31st Mar
```

Comment:
Is there any need for a fast path for negative ``daysToAdd``? Or is it mostly used for moving forwards?